### ISSUE TITLE
Implement smart filtering of join/part/quit messages

### ISSUE THREAD

**Issue by dbkaplun** [erming/shout#432]
Please add an option so that only join/part/quit messages are shown from people who have recently spoken.

**Comment by dmp1ce** [comment]
This would be a nice feature. It is a feature I miss from using WeeChat.

**Comment by dubzi** [comment]
Another option would be to have leave/join messages combined like IRCCloud does. After a few of them happened with no chat between them, they get combined into a single line instead of all taking up their own line.

**Comment by YaManicKill** [comment]
@dubzi Yeah, I think if #48 was done, then this would probably not be needed. @dbkaplun @dmp1ce if #48 was implemented, would you still want this, or would that override the need for this?

**Comment by dmp1ce** [comment]
I'd be happy with either one. The problem for me is join/part/quit taking up so much space when all I really want to see is the chat.

I think I favor smart filtering over combining chats a little because on large channels you can get a join/part/quit in between every message when they are probably irrelevant.
